src/views/PackageLatest/index.tsx

```diff
@@ -7,11 +11,41 @@ interface RouteParams {
 
 export function PackageLatest() {
   const { name, scope }: RouteParams = useParams();
-  const version = "latest"; // read latest version from state here
+  const catalog = useCatalog();
 
+  if (catalog.loading || !catalog.data) {
+    return (
+      <Center minH="200px">
+        <Spinner size="xl" />
+      </Center>
+    );
+  }
+
+  return <Redirect to={buildRedirectUrl(catalog.data, name, scope)} />;
+}
+
+export function buildRedirectUrl(
+  catalog: Packages,
+  name: string,
+  scope?: string
+) {
   const prefix = "/packages/";
-  const packagePath = scope ? `${scope}/${name}` : `${name}`;
+  const packageName = getFullPackageName(name, scope);
+  const version = findPackage(catalog, packageName).version;
   const suffix = `/v/${version}`;
+  return `${prefix}${packageName}${suffix}`;
+}
+
+function findPackage(catalog: Packages, pkg: string) {
+  const packages = catalog.packages.filter((p) => p.name === pkg);
+
+  if (packages.length === 0) {
+    throw new Error(`Package ${pkg} does not exist in catalog`);
+  }
+
+  if (packages.length > 1) {
+    throw new Error(`Multiple packages found for ${pkg} in catalog`);
+  }
 
-  return <Redirect to={`${prefix}${packagePath}${suffix}`} />;
+  return packages[0];
```

Comment:
going along with my comment from above, we could just do something like

```ts
const package = catalog.packages.find(({ name }) => name === pkg);

if (!package) {
  throw new Error(...);
}

return package;
```

Reply:
https://github.com/cdklabs/construct-hub-webapp/pull/137#discussion_r658056724